File: lib/features/attachments/domain/attachment.dart
/// A patient document attachment (УЗИ-заключение, анализ на ВИЧ, прочие сканы).
/// Mirrors the backend `AttachmentOut`. Bytes are fetched separately via
/// `GET /attachments/{id}/file`.
class Attachment {
  const Attachment({
    required this.id,
    required this.patientId,
    required this.kind,
    required this.createdAt,
    this.visitId,
    this.operationId,
    this.originalName,
    this.contentType,
    this.size,
    this.note,
    this.uploadedById,
    this.uploadedByName,
  });

  final String id;
  final String patientId;
  final String kind; // uzi | hiv | lab | other
  final String createdAt;
  final String? visitId;
  final String? operationId;
  final String? originalName;
  final String? contentType;
  final int? size;
  final String? note;
  final String? uploadedById;
  final String? uploadedByName;

  factory Attachment.fromJson(Map<String, dynamic> json) => Attachment(
        id: json['id'] as String,
        patientId: json['patient_id'] as String,
        kind: json['kind'] as String? ?? 'other',
        createdAt: json['created_at'] as String? ?? '',
        visitId: json['visit_id'] as String?,
        operationId: json['operation_id'] as String?,
        originalName: json['original_name'] as String?,
        contentType: json['content_type'] as String?,
        size: (json['size'] as num?)?.toInt(),
        note: json['note'] as String?,
        uploadedById: json['uploaded_by_id'] as String?,
        uploadedByName: json['uploaded_by_name'] as String?,
      );

  static const kindLabels = <String, String>{
    'uzi': 'УЗИ',
    'hiv': 'Анализ на ВИЧ',
    'lab': 'Лаборатория',
    'other': 'Документ',
  };

  String get kindLabel => kindLabels[kind] ?? 'Документ';

  String get displayName =>
      (originalName != null && originalName!.trim().isNotEmpty)
          ? originalName!
          : '$kindLabel.pdf';

  String get dateLabel =>
      createdAt.isEmpty ? '' : createdAt.replaceFirst('T', ' ').split('.').first;
}

File: backend/app/schemas/attachment.py
"""DTOs for patient file attachments (УЗИ, анализ на ВИЧ, прочие документы)."""
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AttachmentOut(BaseModel):
    """A stored document on the patient record. Bytes are fetched separately via
    ``GET /attachments/{id}/file`` (auth-gated) — this carries only metadata."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    patient_id: UUID
    visit_id: UUID | None = None
    operation_id: UUID | None = None
    kind: str
    original_name: str | None = None
    content_type: str | None = None
    size: int | None = None
    note: str | None = None
    cabinet: str | None = None        # room the study was done in
    uploaded_by_id: UUID | None = None
    uploaded_by_name: str | None = None
    created_at: datetime
